### ISSUE TITLE
tensorflow 1.14 build failed 

### ISSUE THREAD

**Issue by chengleiwang** [tensorflow/tensorflow#31434]
**System information**
- OS Platform and Distribution (Linux Ubuntu 18.04):
- TensorFlow installed from (source):
- TensorFlow version: r1.14
- Python version: v3.6.8
- Bazel version (tried v0.24.1, v0.26, v0.28):
- GCC/Compiler version (gcc version 7.4):
- CUDA/cuDNN version: 10.0/7.5
- GPU model and memory: GTX1050

**Describe the problem**
I tried to use below command to build concat_op_test application.
`bazel test -c opt  --strip=never  -s //tensorflow/core/kernels:concat_op_test --verbose_failures` 
The compiler I select is clang and build with CUDA support.
I got below errors when bazel version is 0.24.1 and 0.26
```
ERROR: Skipping '//tensorflow/core/kernels:concat_op_test': error loading package 'tensorflow/core/kernels': Encountered error while reading extension file 'cu
da/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
        File "/home/chengleiwang/code/tensorflow/third_party/gpus/cuda_configure.bzl", line 1266
                _create_local_cuda_repository(repository_ctx)
        File "/home/chengleiwang/code/tensorflow/third_party/gpus/cuda_configure.bzl", line 1199, in _create_local_cuda_repository
                _tpl(repository_ctx, "crosstool:BUILD", c...)
        File "/home/chengleiwang/code/tensorflow/third_party/gpus/cuda_configure.bzl", line 745, in _tpl
                repository_ctx.template(out, Label(("//third_party/gpus/%s...)), ...)
class com.google.devtools.build.lib.syntax.SkylarkList$MutableList cannot be cast to class java.lang.String (com.google.devtools.build.lib.syntax.SkylarkList$MutableList is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
ERROR: error loading package 'tensorflow/core/kernels': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cu
da//cuda': Traceback (most recent call last):
        File "/home/chengleiwang/code/tensorflow/third_party/gpus/cuda_configure.bzl", line 1266
                _create_local_cuda_repository(repository_ctx)
        File "/home/chengleiwang/code/tensorflow/third_party/gpus/cuda_configure.bzl", line 1199, in _create_local_cuda_repository
                _tpl(repository_ctx, "crosstool:BUILD", c...)
        File "/home/chengleiwang/code/tensorflow/third_party/gpus/cuda_configure.bzl", line 745, in _tpl
                repository_ctx.template(out, Label(("//third_party/gpus/%s...)), ...)
class com.google.devtools.build.lib.syntax.SkylarkList$MutableList cannot be cast to class java.lang.String (com.google.devtools.build.lib.syntax.SkylarkList$MutableList is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
``` 
I got below errors when bazel version is 0.28.0 (I changed the bazel version check in configure.py   `current_bazel_version = check_bazel_version('0.24.1', '0.28.2')` )
```
Starting local Bazel server and connecting to it...
Internal error thrown during build. Printing stack trace: java.lang.RuntimeException: Unrecoverable error while evaluating node 'REPOSITORY_DIREC
TORY:@local_config_cuda' (requested by nodes 'REPOSITORY:@local_config_cuda')
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class com.google.devtools.build.lib.syntax.SkylarkList$MutableList cannot be cast to class java.lang.String (com.google.devtools.build.lib.syntax.SkylarkList$MutableList is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
        at com.google.devtools.build.lib.bazel.repository.skylark.SkylarkRepositoryContext.createFileFromTemplate(SkylarkRepositoryContext.java:295)
        at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
        at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.syntax.FuncallExpression.callFunction(FuncallExpression.java:992)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:904)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.syntax.FuncallExpression.callFunction(FuncallExpression.java:992)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:904)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.Eval.execStatements(Eval.java:231)
        at com.google.devtools.build.lib.syntax.Eval.execIf(Eval.java:144)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:214)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.bazel.repository.skylark.SkylarkRepositoryFunction.fetch(SkylarkRepositoryFunction.java:167)
        at com.google.devtools.build.lib.rules.repository.RepositoryDelegatorFunction.fetchRepository(RepositoryDelegatorFunction.java:288)
        at com.google.devtools.build.lib.rules.repository.RepositoryDelegatorFunction.compute(RepositoryDelegatorFunction.java:215)
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:451)
        ... 4 more

INFO: Elapsed time: 2.858s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: tensorflow/core/kernels
    Fetching @local_config_cuda; fetching
Internal error thrown during build. Printing stack trace: java.lang.RuntimeException: Unrecoverable error while evaluating node 'REPOSITORY_DIRECTORY:@local_config_cuda' (requested by nodes 'REPOSITORY:@local_config_cuda')
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class com.google.devtools.build.lib.syntax.SkylarkList$MutableList cannot be cast to class java.lang.String (com.google.devtools.build.lib.syntax.SkylarkList$MutableList is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
        at com.google.devtools.build.lib.bazel.repository.skylark.SkylarkRepositoryContext.createFileFromTemplate(SkylarkRepositoryContext.java:295)
        at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
        at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.syntax.FuncallExpression.callFunction(FuncallExpression.java:992)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:904)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.syntax.FuncallExpression.callFunction(FuncallExpression.java:992)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:904)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.Eval.execStatements(Eval.java:231)
        at com.google.devtools.build.lib.syntax.Eval.execIf(Eval.java:144)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:214)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.bazel.repository.skylark.SkylarkRepositoryFunction.fetch(SkylarkRepositoryFunction.java:167)
        at com.google.devtools.build.lib.rules.repository.RepositoryDelegatorFunction.fetchRepository(RepositoryDelegatorFunction.java:288)
        at com.google.devtools.build.lib.rules.repository.RepositoryDelegatorFunction.compute(RepositoryDelegatorFunction.java:215)
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:451)
        ... 4 more
java.lang.RuntimeException: Unrecoverable error while evaluating node 'REPOSITORY_DIRECTORY:@local_config_cuda' (requested by nodes 'REPOSITORY:@local_config_cuda')
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class com.google.devtools.build.lib.syntax.SkylarkList$MutableList cannot be cast to class java.lang.String (com.google.devtools.build.lib.syntax.SkylarkList$MutableList is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
        at com.google.devtools.build.lib.bazel.repository.skylark.SkylarkRepositoryContext.createFileFromTemplate(SkylarkRepositoryContext.java:295)
        at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
        at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.syntax.FuncallExpression.callFunction(FuncallExpression.java:992)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:904)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.syntax.FuncallExpression.callFunction(FuncallExpression.java:992)
        at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:904)
        at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:201)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.Eval.execStatements(Eval.java:231)
        at com.google.devtools.build.lib.syntax.Eval.execIf(Eval.java:144)
        at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:214)
        at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
        at com.google.devtools.build.lib.syntax.UserDefinedFunction.call(UserDefinedFunction.java:91)
        at com.google.devtools.build.lib.syntax.BaseFunction.callWithArgArray(BaseFunction.java:475)
        at com.google.devtools.build.lib.syntax.BaseFunction.call(BaseFunction.java:437)
        at com.google.devtools.build.lib.bazel.repository.skylark.SkylarkRepositoryFunction.fetch(SkylarkRepositoryFunction.java:167)
        at com.google.devtools.build.lib.rules.repository.RepositoryDelegatorFunction.fetchRepository(RepositoryDelegatorFunction.java:288)
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: tensorflow/core/kernels
    Fetching @local_config_cuda; fetching
```

**Comment by chengleiwang** [comment]
This version could build pass:  c1de6bc140d22b2b646d8e0f925ed7dc0bf21a13
Any regression?

**Comment by angersson** [comment]
@oanush, I'm afraid I lack expertise with this part of the code. Can you find someone more appropriate to take a look?

**Comment by mihaimaruseac** [comment]
Can you try a new checkout of the branch? There was a recent cherry-pick regarding java.

**Comment by chengleiwang** [comment]
Try the newest checkout of r1.14 ，still same error...
```
commit e19c354920c3b246dda6598229210a582caaa1a9 (HEAD -> r1.14, origin/r1.14)
Author: Mihai Maruseac <mihaimaruseac@google.com>
Date:   Thu Aug 8 08:33:18 2019 -0700

    Fix sanity build failure after #31425
    
    Error that this is fixing is
    
    ```
    tensorflow/python/framework/func_graph.py:32: [W0611(unused-import), ] Unused composite_tensor imported from tensorflow.python.framework
```

**Comment by mihaimaruseac** [comment]
Unfortunately I don't have ideas on how to debug Java, adding @sjamesr

**Comment by chengleiwang** [comment]
Ping @sjamesr , any suggestion?

**Comment by mengdong** [comment]
I recommend to try gcc 6.5, you can use update-alternatives to switch between gcc 7.4 and gcc 6.5. The build should be successful. @chengleiwang

**Comment by chengleiwang** [comment]
@mengdong, actually I am using clang as compiler.  So how could update-alternatives  gcc can build successful?

**Comment by mengdong** [comment]
ah, sorry I missed, I haven't built with clang before hence no comment on that.

**Comment by mihaimaruseac** [comment]
Afaik update-alternatives can also work with clang. Though I don't know which version is the one that's supported best

**Comment by SamuelMarks** [comment]
Confirmed same error on Ubuntu 18.04 with Bazel 0.25.2, clang 6.0.0-1ubuntu2, Python 3.6.8, NVIDIA Tesla K80, CUDA Version: 10.2, with v1.14.0 checked out (@ 87989f69597d6b2d60de8f112e1e3cea23be7298)

**Comment by mihaimaruseac** [comment]
Can you try `r1.15` branch instead? Turns out our 1.14 branch is in a terrible state, sorry for that.

**Comment by SamuelMarks** [comment]
Confirmed `master` branch breaks also: https://github.com/tensorflow/tensorflow/issues/32323

I can try with `r1.15` next

**Comment by SamuelMarks** [comment]
Confirmed errors with `r1.15` (ea930781c3164c9646e26cf3716f86804aa65b63):
```
ERROR: /home/ubuntu/.cache/bazel/_bazel_ubuntu/7d3379297965407e4f621f04220127eb/external/nccl_archive/BUILD.bazel:53:1: output 'external/nccl_archive/_objs/device_lib/functions.cu.o' was not created
[4,495 / 6,642] 4 actions running
    Compiling external/nccl_archive/src/collectives/device/functions.cu.cc [for host]; 1s local
    Compiling external/nccl_archive/src/collectives/device/max_f64_all_reduce.cu.cc [for host]; 0s local
    Compiling external/nccl_archive/src/collectives/device/min_u8_all_reduce.cu.cc [for host]; 0s local
    Compiling external/nccl_archive/src/collectives/device/sum_i8_all_gather.cu.cc [for host]; 0s local
ERROR: /home/ubuntu/.cache/bazel/_bazel_ubuntu/7d3379297965407e4f621f04220127eb/external/nccl_archive/BUILD.bazel:53:1: not all outputs were created or valid
[4,495 / 6,642] 4 actions running
    Compiling external/nccl_archive/src/collectives/device/functions.cu.cc [for host]; 1s local
    Compiling external/nccl_archive/src/collectives/device/max_f64_all_reduce.cu.cc [for host]; 0s local
    Compiling external/nccl_archive/src/collectives/device/min_u8_all_reduce.cu.cc [for host]; 0s local
    Compiling external/nccl_archive/src/collectives/device/sum_i8_all_gather.cu.cc [for host]; 0s local
Target //tensorflow/tools/pip_package:build_pip_package failed to build
[4,499 / 6,642] checking cached actions
Use --verbose_failures to see the command lines of failed build steps.
[4,499 / 6,642] checking cached actions
INFO: Elapsed time: 416.926s, Critical Path: 54.00s
[4,499 / 6,642] checking cached actions
INFO: 453 processes: 453 local.
[4,499 / 6,642] checking cached actions
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

With an earlier build failing on:
```
[3,141 / 3,324] 4 actions running
    Compiling tensorflow/core/kernels/spacetobatch_functor_gpu.cu.cc [for host]; 13s local
    Compiling tensorflow/core/kernels/reverse_sequence_op.cc [for host]; 3s local
    Compiling tensorflow/core/kernels/one_hot_op_gpu.cu.cc [for host]; 2s local
    Compiling tensorflow/core/kernels/reverse_op.cc [for host]; 0s local
ERROR: /home/ubuntu/repos/tensorflow-for-py3/tensorflow/core/kernels/BUILD:4651:1: output 'tensorflow/core/kernels/_objs/batch_space_ops_gpu/spacetobatch_functor_gpu.cu.pic.o' was not created
[3,141 / 3,324] 4 actions running
    Compiling tensorflow/core/kernels/spacetobatch_functor_gpu.cu.cc [for host]; 13s local
    Compiling tensorflow/core/kernels/reverse_sequence_op.cc [for host]; 3s local
    Compiling tensorflow/core/kernels/one_hot_op_gpu.cu.cc [for host]; 2s local
    Compiling tensorflow/core/kernels/reverse_op.cc [for host]; 0s local
ERROR: /home/ubuntu/repos/tensorflow-for-py3/tensorflow/core/kernels/BUILD:4651:1: not all outputs were created or valid37.198.227] out: 
[3,141 / 3,324] 4 actions running
    Compiling tensorflow/core/kernels/spacetobatch_functor_gpu.cu.cc [for host]; 13s local
    Compiling tensorflow/core/kernels/reverse_sequence_op.cc [for host]; 3s local
    Compiling tensorflow/core/kernels/one_hot_op_gpu.cu.cc [for host]; 2s local
    Compiling tensorflow/core/kernels/reverse_op.cc [for host]; 0s local
Target //tensorflow/tools/pip_package:build_pip_package failed to build
[3,145 / 3,324] checking cached actions
Use --verbose_failures to see the command lines of failed build steps.
[3,145 / 3,324] checking cached actions
INFO: Elapsed time: 317.459s, Critical Path: 116.62s
[3,145 / 3,324] checking cached actions
INFO: 157 processes: 157 local.
[3,145 / 3,324] checking cached actions
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

**Comment by mihaimaruseac** [comment]
All I can think of for next step is to try building from scratch again (make sure on clean environment) and also pass in `--verbose_failures` flag when building.

Then, please _attach_ a log of the entire compilation.

**Comment by SamuelMarks** [comment]
Full output (with input and environment variables also shown): https://gist.github.com/bc5e1baf640afa3acd54a0ed0b4a0301

Ran this beforehand:
```
rm -rf ~/repos/tensorflow-for-py3 ~/.cache/bazel
```

**Comment by mihaimaruseac** [comment]
So there are 36 lines of

```
out: ./tensorflow/core/util/gpu_device_functions.h(648): error: calling a constexpr __host__ function("real") from a __device__ function("GpuAtomicAdd") is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.
```

**Comment by SamuelMarks** [comment]
So you're asking me to run the build again, this time with `--expt-relaxed-constexpr`?

**Comment by SamuelMarks** [comment]
Hmm:
```
bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package --verbose_failures --expt-relaxed-constexpr
out: Starting local Bazel server and connecting to it...
out: ERROR: Unrecognized option: --expt-relaxed-constexpr
```

**Comment by chengleiwang** [comment]
> Hmm:
> 
> ```
> bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package --verbose_failures --expt-relaxed-constexpr
> out: Starting local Bazel server and connecting to it...
> out: ERROR: Unrecognized option: --expt-relaxed-constexpr
> ```

I also met this error before. I tried to add --expt-relaxed-constexpr in tensorflow.bzl _cuda_copts() function. but still with no lucky.

```
def _cuda_copts(opts = []):
    """Gets the appropriate set of copts for (maybe) CUDA compilation.

      If we're doing CUDA compilation, returns copts for our particular CUDA
      compiler.  If we're not doing CUDA compilation, returns an empty list.

      """
    return cuda_default_copts() + select({
        "//conditions:default": [],
        "@local_config_cuda//cuda:using_nvcc": ([
            "-nvcc_options=relaxed-constexpr",
            "-nvcc_options=ftz=true",
        ]),
        "@local_config_cuda//cuda:using_clang": ([
            "-fcuda-flush-denormals-to-zero",
            "--expt-relaxed-constexpr",
        ]),
    }) + if_cuda_is_configured_compat(opts)
```

**Comment by mihaimaruseac** [comment]
I think you have to use the `copt` option in Bazel.

**Comment by SamuelMarks** [comment]
Overnight I ran:
```
bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package --verbose_failures --copt='--expt-relaxed-constexpr'
```

But looking at the log, it's still littered with:
```
./tensorflow/core/kernels/reduction_gpu_kernels.cu.h(121): warning: calling a constexpr __host__ function("imag") from a __host__ __device__ function("operator()") is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.
```

FYI: Build it still failing, on r1.15 branch @ 2e29d3b35a4c5363258ad040a12b3a15affff3c2

**Comment by mohantym** [comment]
Hi @chengleiwang !We see that you are using old version of Tensorflow which is officially considered as end of life, We recommend that you upgrade to 2.6 version and let us know if the issue still persists in newer versions .Please open a new issue in case you face any errors, we will get you the right help .Thanks!
